File: src/features/work-hours/domain/repositories/task-preset-repository.ts
import type {
  TaskPreset,
  TaskPresetInput,
} from "@/features/work-hours/domain/entities/work-hours";

export interface TaskPresetRepository {
  listByProfile(profileId: string): Promise<TaskPreset[]>;
  create(profileId: string, input: TaskPresetInput): Promise<TaskPreset>;
  update(id: string, input: TaskPresetInput): Promise<TaskPreset>;
  remove(id: string): Promise<void>;
}

File: src-tauri/src/domain/repositories/task_preset_repository.rs
use async_trait::async_trait;

use crate::domain::entities::task_preset::TaskPreset;
use crate::domain::error::DomainError;

/// Persistence contract for saved task presets (implemented over the unlocked
/// vault connection).
#[async_trait]
pub trait TaskPresetRepository: Send + Sync {
    async fn create(&self, preset: &TaskPreset) -> Result<(), DomainError>;

    async fn list_by_profile(&self, profile_id: &str) -> Result<Vec<TaskPreset>, DomainError>;

    async fn find_by_id(&self, id: &str) -> Result<Option<TaskPreset>, DomainError>;

    async fn update(&self, preset: &TaskPreset) -> Result<(), DomainError>;

    async fn delete(&self, id: &str) -> Result<(), DomainError>;
}
